### ISSUE TITLE
Compile failure on Darwin

### ISSUE THREAD

**Issue by aaronscode** [rust-motd/rust-motd#20]
As seen in [this issue for NixOs](https://github.com/NixOS/nixpkgs/pull/153860) compillation fails for Mac. The partial log reveals the issue to be with the systemstat crate:

```
  |
59 |                 .meminfo
   |                  ^^^^^^^ unknown field
   |
   = note: available fields are: `active`, `inactive`, `wired`, `cache`, `free`

For more information about this error, try `rustc --explain E0609`.
error: could not compile `rust-motd` due to 2 previous errors
builder for '/nix/store/bjacpn3wsxavnp362dp86vnk8z7v69bm-rust-motd-0.2.0.drv' failed with exit code 101
error: build of '/nix/store/bjacpn3wsxavnp362dp86vnk8z7v69bm-rust-motd-0.2.0.drv' failed
```

Apparently it implements different fields for different OS's using conditional compilation ([e.g.](https://github.com/unrelentingtech/systemstat/blob/ad3a0a2c1e0577dd28b9a653d352f31a34528123/src/data.rs#L40)). 

A solution would be to add conditional compilation of our own to reference the available fields when compiling for Mac, and test that the compilation succeeds when cross compiling for the x86_64 and aarch64 darwin targets.

**Comment by NickLarsenNZ** [comment]
Shouldn't we use the agnostic `Memory` struct, instead of `PlatformMemory`?

https://github.com/unrelentingtech/systemstat/blob/ad3a0a2c1e0577dd28b9a653d352f31a34528123/src/data.rs#L237-L243

**Comment by MarcelRobitaille** [comment]
@NickLarsenNZ I believe PlatformMemory was used in order to get info about the swap usage in addition to RAM (please correct me if I'm wrong @aaronscode).

I know Windows uses a pagefile, which is basically equivalent to swap. I'm not sure what we could use for macOS. The available fields are `active`, `wired`, `cache`, and `free`: https://github.com/unrelentingtech/systemstat/blob/ad3a0a2c1e0577dd28b9a653d352f31a34528123/src/data.rs#L222-L228. Maybe we could fall back to only showing physical RAM (at least as a workaround to get it working at all on macOS).

**Comment by NickLarsenNZ** [comment]
Ah yeah, just noticed that while I was trying to fix it.

I ran the systemstat example and it seems like it doesn't support macos:
https://github.com/unrelentingtech/systemstat/blob/master/examples/info.rs


```
Mounts:
/dev/disk1s5s1 ---apfs---> / (available 107.9 GB of 500.0 GB)
devfs ---devfs---> /dev (available 0 B of 351.7 KB)
/dev/disk1s4 ---apfs---> /System/Volumes/VM (available 108.0 GB of 500.0 GB)
/dev/disk1s2 ---apfs---> /System/Volumes/Preboot (available 108.0 GB of 500.0 GB)
/dev/disk1s6 ---apfs---> /System/Volumes/Update (available 108.0 GB of 500.0 GB)
/dev/disk1s1 ---apfs---> /System/Volumes/Data (available 107.9 GB of 500.0 GB)
map auto_home ---autofs---> /System/Volumes/Data/home (available 0 B of 0 B)
/dev/disk1s5 ---apfs---> /System/Volumes/Update/mnt1 (available 108.0 GB of 500.0 GB)

Mount at /:
/dev/disk1s5s1 ---apfs---> / (available 107.9 GB of 500.0 GB)

Block statistics error: Not supported

Networks:
awdl0 ([NetworkAddrs { addr: V6(fe80::..............:77ee), netmask: V6(ffff:ffff:ffff:ffff::) }])
bridge0 ([])
en0 ([NetworkAddrs { addr: V6(fe80::..............:308d), netmask: V6(ffff:ffff:ffff:ffff::) }, NetworkAddrs { addr: V4(192.168.178.23), netmask: V4(255.255.255.0) }, NetworkAddrs { addr: V6(2003:..............:723e), netmask: V6(ffff:ffff:ffff:ffff::) }, NetworkAddrs { addr: V6(2003:..............:aac6), netmask: V6(ffff:ffff:ffff:ffff::) }])
en1 ([])
en2 ([])
en3 ([])
en4 ([])
en5 ([NetworkAddrs { addr: V6(fe80::..............:1122), netmask: V6(ffff:ffff:ffff:ffff::) }])
gif0 ([])
llw0 ([NetworkAddrs { addr: V6(fe80::..............:77ee), netmask: V6(ffff:ffff:ffff:ffff::) }])
lo0 ([NetworkAddrs { addr: V4(127.0.0.1), netmask: V4(255.0.0.0) }, NetworkAddrs { addr: V6(::1), netmask: V6(ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff) }, NetworkAddrs { addr: V6(fe80::1), netmask: V6(ffff:ffff:ffff:ffff::) }])
p2p0 ([])
stf0 ([])
utun0 ([NetworkAddrs { addr: V6(fe80::..............:e8f0), netmask: V6(ffff:ffff:ffff:ffff::) }])
utun1 ([NetworkAddrs { addr: V6(fe80::..............:807c), netmask: V6(ffff:ffff:ffff:ffff::) }])
utun2 ([NetworkAddrs { addr: V6(fe80::..............:69e), netmask: V6(ffff:ffff:ffff:ffff::) }])

Network interface statistics:
awdl0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
bridge0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
en0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
en1 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
en2 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
en3 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
en4 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
en5 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
gif0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
llw0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
lo0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
p2p0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
stf0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
utun0 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
utun1 statistics: (Err(Custom { kind: Other, error: "Not supported" }))
utun2 statistics: (Err(Custom { kind: Other, error: "Not supported" }))

Battery: error: Not supported, AC power: error: Not supported

Memory: error: Not supported

Load average: 5.0424805 6.515625 5.930176

Uptime: 68286.759229657s

Boot time: 2022-04-02 13:41:42.000835343 UTC

CPU load: error: Not supported

CPU temp: Not supported

System socket statistics: error: Not supported
```
